### ISSUE TITLE
Improving flow for entering project address on create project 

### ISSUE THREAD

**Issue by divine-comedian** [Giveth/giveth-dapps-v2#3078]
This is the current flow:

https://github.com/Giveth/giveth-dapps-v2/assets/67759413/78235bf0-ac40-4da4-a985-e09635b8b38a

Currently the user needs to click Add Address -> Paste Address -> Click Save for each address on every chain 

We should not have so many clicks and so many pop ups just to add a single address! 

Let's try to simplify this: 
- Create a text field for a user to enter an address that will be set for all chains, user can paste the address in directly - they click a button 'Apply' - and it is set for all chains below. 
- Copy this warning and place it underneath the text field: 
> Please DO NOT enter an exchange deposit address, use a key you control in a wallet like Metamask, or your funds maybe lost
- Check the address is valid for all chains and show error text underneath the text field, but above the warning message if there are any problems (i.e bad formatting, address is duplicate or ineligible) 
- User cannot publish if there are any errors
- User can still manually edit each chain address like normal even after clicking apply button as described above.

WDYT @MoeNick ?

**Comment by mosaeedi** [comment]
The design for this issue is done.

You can see how it works here -> [Youtube video](https://youtu.be/wRbHavTb1PI)

You can see all of it's different states on Figma -> [Wallet component on Figma](https://www.figma.com/file/vMiyVd1Ly5LjgSyZrChVV8/Giveth-users?type=design&node-id=1327-6201&mode=design&t=0HE5MaJjnZ8vBgEc-4)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNzBSQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c0c9a9f282d1b2da5ef82f5dc50dc1cf295770a1/Screenshot%202023-11-28%20at%203.09.07%E2%80%AFPM.png)


@divine-comedian

**Comment by divine-comedian** [comment]
@mosaeedi 

Left some comments in the figma 

I'm not a big fan of this modal:
![image](https://github.com/Giveth/giveth-dapps-v2/assets/67759413/fa1bc4aa-573c-456b-8435-d779f7cee9b4)

But I'm okay if you think we need it.

**Comment by MoeNick** [comment]
The fields of Im receiving fund on this network is now history, we removed the functionality of this binary input, whether it has value or not, if it has we assume it receiving funds on this network. so I think it makes it complicated, lets remove that check box.

**Comment by divine-comedian** [comment]
> The fields of Im receiving fund on this network is now history, we removed the functionality of this binary input, whether it has value or not, if it has we assume it receiving funds on this network. so I think it makes it complicated, lets remove that check box.

So if the user would not like to receive funds on a given network then you suggest they just leave that field blank for the given network?

**Comment by MoeNick** [comment]
> > The fields of Im receiving fund on this network is now history, we removed the functionality of this binary input, whether it has value or not, if it has we assume it receiving funds on this network. so I think it makes it complicated, lets remove that check box.
> 
> So if the user would not like to receive funds on a given network then you suggest they just leave that field blank for the given network?

Yes @divine-comedian 
this is  Dapp logic on frontend to recognize this address is active or not, not the boolean field.

**Comment by divine-comedian** [comment]
> > > The fields of Im receiving fund on this network is now history, we removed the functionality of this binary input, whether it has value or not, if it has we assume it receiving funds on this network. so I think it makes it complicated, lets remove that check box.
> > 
> > 
> > So if the user would not like to receive funds on a given network then you suggest they just leave that field blank for the given network?
> 
> Yes @divine-comedian this is Dapp logic on frontend to recognize this address is active or not, not the boolean field.

I agree that this is less complex and makes less dev work

**Comment by divine-comedian** [comment]
Need to decide when we check for the validity of the addresses 

when user submits? did we get rate limited

**Comment by divine-comedian** [comment]
@mosaeedi 

What should we do here? 
![image](https://github.com/Giveth/giveth-dapps-v2/assets/67759413/43ec5cf7-db2b-4763-ba43-3844d0ebdcbb)

**Comment by divine-comedian** [comment]
@alireza-sharifpour 

Thanks for catching this one! I think we can start it - This is to modify the view for adding project recipient addresses on create project and edit project 

Check out the video Mo posted to see more. The most important change is removing the "Don't accept funds on this chain" checkbox which has already been reflected on the design, If a user does not want to receive funds on x chain.

Three questions you can help answer:

- [ ] When should we check for the above validations? ENS, valid address etc.. Can we check after a short delay of updates to the state? i.e once the state of the input box changes we validate it after a 1.5 second delay or..?
- [ ] When a recipient address is already taken by another project - on the error modal is it easy to find out which project is using the address and add it to the error toast notification?
- [ ] On what chains can the dapp handle resolving ENS addresses? Only mainnet?

**Comment by alireza-sharifpour** [comment]
Hey @divine-comedian,

Took a look at the new wallet address design. It's kinda reminiscent of our setup four months ago, but now with an "Apply Address to All" button. Remember why we shifted to the modal? Maybe @MoeNick knows.

Thinking about reverting means redoing all the validation, state management, and error handling we tweaked during that modal switch. It's a bit of a maze to backtrack through. Any strong reasons for going back?

Also, a quick heads up on the mobile side—button-in-textbox might not play nice on smaller screens.

About the questions.
1- Yes we can.
2- For showing which project using this address, we need backend changes.
3- Only Mainnet (with my current knowledge).
